features/iotserver-feature/org.wso2.carbon.device.mgt.iot.feature/src/main/resources/jaggeryapps/store/extensions/app/store-device-type/themes/store/js/policy-add-new.js

```diff
@@ -0,0 +1,225 @@
+                        updateGroupedInputVisibility(this);
+                    });
+                });
+            },
+            250 // time delayed for the execution of above function, 250 milliseconds
+        );
+    };
+
+
+    /**
+     * Method to update the visibility of grouped input.
+     * @param domElement HTML grouped-input element with class name "grouped-input"
+     */
+    var updateGroupedInputVisibility = function (domElement) {
+        if ($(".parent-input:first", domElement).is(":checked")) {
+            if ($(".grouped-child-input:first", domElement).hasClass("disabled")) {
+                $(".grouped-child-input:first", domElement).removeClass("disabled");
+            }
+            $(".child-input", domElement).each(function () {
+                $(this).prop('disabled', false);
+            });
+        } else {
+            if (!$(".grouped-child-input:first", domElement).hasClass("disabled")) {
+                $(".grouped-child-input:first", domElement).addClass("disabled");
+            }
+            $(".child-input", domElement).each(function () {
+                $(this).prop('disabled', true);
+            });
+        }
+    };
+
+    stepperRegistry['policy-content'] = function (actionButton) {
+        policy.policyName = $("#policy-name-input").val();
+        policy.policyDescription = $("#policy-description-input").val();
+    };
+
+    stepperRegistry['policy-profile'] = function (actionButton) {
+        policy.policyDefinition = window.queryEditor.getValue();
+        //All data is collected. Policy can now be created.
+        savePolicy();
+    };
+
+    $(".uu").click(function () {
+        var policyName = $("#policy-name-input").val();
+        var selectedProfiles = $("#profile-input").find(":selected");
+        var selectedProfileId = selectedProfiles.data("id");
+        var selectedUserRoles = $("#user-roles-input").val();
+        var selectedUsers = $("#users-input").val();
+        var selectedAction = $("#action-input").val();
+    });
+
+    var mime = MIME_TYPE_SIDDHI_QL;
+
+    // get mime type
+    if (window.location.href.indexOf('mime=') > -1) {
+        mime = window.location.href.substr(window.location.href.indexOf('mime=') + 5);
+    }
+
+    window.queryEditor = CodeMirror.fromTextArea(document.getElementById('policy-definition-input'), {
+        mode: mime,
+        indentWithTabs: true,
+        smartIndent: true,
+        lineNumbers: true,
+        matchBrackets: true,
+        autofocus: true,
+        extraKeys: {
+            "Shift-2": function (cm) {
+                insertStr(cm, cm.getCursor(), '@');
+                CodeMirror.showHint(cm, getAnnotationHints);
+            },
+            "Ctrl-Space": "autocomplete"
+        }
+    });
+
+});
+
+function getQueryParams(key) {
+    var qs = document.location.search.split('+').join(' ');
+
+    var params = {},
+            tokens,
+            re = /[?&]?([^=]+)=([^&]*)/g;
```

Comment:
inappropriate line breaks for one statement. also use meaningful names for variables -> ex: qs & re